packages/svelte/src/performance.ts

```diff
@@ -0,0 +1,93 @@
+import { getCurrentHub } from '@sentry/browser';
+import { Span, Transaction } from '@sentry/types';
+import { afterUpdate, beforeUpdate, onMount } from 'svelte';
+import { current_component } from 'svelte/internal';
+
+import { DEFAULT_COMPONENT_NAME, UI_SVELTE_INIT, UI_SVELTE_UPDATE } from './constants';
+import { TrackComponentOptions } from './types';
+
+const defaultOptions: Required<Pick<TrackComponentOptions, 'trackInit' | 'trackUpdates'>> &
+  Pick<TrackComponentOptions, 'componentName'> = {
```

Comment:
Subjective: I'd rather have the possible options written out right here instead of using these indirections via `Pick`. It's just way easier to build a mental model of what's going on than having to go through three type definitions to figure out what type `componentName` is supposed to be.

There is the argument that doing it this way makes it easier when we decide to change the underlying type, but that is not true IMO. When we change the underlying type, we have to go in and change the non-type code anyways so there is not much to be gained here. Rant over lol

Reply:
I think there's cases where Pick is a good/better choice (and we use it in multiple places in the SDK) but in this case I changed it this type signature is really a little complicated for what it does. My goal was just to make it explicitly clear that the default options set `trackInit` and `trackUpdates` so that we don't have to check their definedness after merging the options.

Reply:
Way cleaner IMO :D